### PR TITLE
Add initial command handler

### PR DESCRIPTION
**Pull request by pieper126**
What:
Sets up an initial command handler.

Why:
commands should be able to be processed when received from mqtt.

Tests:
Added tests.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Lint with pylint
+        run: |
+          pylint -E --rcfile=./.pylintrc app/ main.py
+      - name: Test with pytest
+        run: |
+          pytest
```
Comment (serroba) [inline .github/workflows/ci.yaml:28]: https://medium.com/@alexey.inkin/how-to-force-newline-at-end-of-files-and-why-you-should-do-it-fdf76d1d090e
https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

Comment (serroba) [reply .github/workflows/ci.yaml:28]: please configure your IDE, so you always include the EOL

File: main.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import os
+import json
+
+from app.modbus_client import ModbusClient
+from app.mqtt_client import MqttClient
+from app.configuration import Configuration
+
+SLAVE=0x01
+
+BROKER_HOST = os.getenv('BROKER_HOST')
```
Comment (serroba) [inline main.py:10]: this should be part of the configuration file, in the YAML file as well.
Essentially everything thats dynamic, should be part of the YAML config

File: main.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+BROKER_PORT = int(os.getenv('BROKER_PORT'))
+MODBUS_HOST = os.getenv('MODBUS_HOST')
+MODBUS_PORT = int(os.getenv('MODBUS_PORT'))
+COMMANDS_TOPIC = os.getenv('COMMANDS_TOPIC')
+CONFIGURATION_PATH = os.getenv('CONFIGURATION_PATH', "config/example_configuration.yaml")
+
+def main():
+    configuration = Configuration.from_file(CONFIGURATION_PATH)
```
Comment (serroba) [inline main.py:18]: We should have a validation step somewhere. To validate that the YAML is in good shape. If is not, explain that is broken

File: main.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    mqtt_client.subscribe_topics([COMMANDS_TOPIC])
+
+    def write_to_modbus(marshalled_message):
+        message = json.loads(marshalled_message)
+        modbus_client.write_coil(message["action"], message["value"])
+
+    mqtt_client.add_message_callback(write_to_modbus)
+
+    mqtt_client.run()
```
Comment (serroba) [inline main.py:30]: this creates a loop,. doesn't it? it keeps listening to new messages.

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+pymodbus
```
Comment (serroba) [inline requirements.txt:1]: not a blocker in this PR, but I feel we should we following the common migration towards poetry. https://evergenworkspace.slack.com/archives/CJE3YMPU7/p1684905901954769

Comment (serroba) [reply requirements.txt:1]: Feel free to address that as a separate piece of work

File: app/configuration.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    def get_coil(self, name: str) -> Coil:
+        return self.coils_map[name]
+    
+    def get_holding_register(self, name: str) -> HoldingRegister:
+        return self.holding_register_map[name]
+    
+def _coils_registers_from_file(path):
+    with open(path, "r", encoding="UTF8") as file:
+        data = yaml.safe_load(file)
+        modbus_mapping = data.get("modbus_mapping", {})
```
Comment (serroba) [inline app/configuration.py:37]: we can split the work here, but as I said bellow, we should have a validation step in place.
I prefer doing that as a separate PR, but I feel that's we should be able to inform to the consumer: _Hey, your config file is missing important things._

File: main.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+from app.configuration import Configuration
+
+SLAVE=0x01
+
+BROKER_HOST = os.getenv('BROKER_HOST')
+BROKER_PORT = int(os.getenv('BROKER_PORT'))
+MODBUS_HOST = os.getenv('MODBUS_HOST')
+MODBUS_PORT = int(os.getenv('MODBUS_PORT'))
+COMMANDS_TOPIC = os.getenv('COMMANDS_TOPIC')
+CONFIGURATION_PATH = os.getenv('CONFIGURATION_PATH', "config/example_configuration.yaml")
```
Comment (serroba) [inline main.py:15]: let's have a default value, but allow to pass by the configuration path as a CLI ARG value.
```
./main.py --config-path /some/path
```

Comment (serroba) [reply main.py:15]: we can also do that in a separate PR though

File: app/mqtt_client.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    
+    def run(self) -> None:
+        """
+        this is a blocking operation.
+        """
+        self.client.on_connect = self._on_connect
+        self.client.on_message = self._on_message
+
+        self.client.connect(self.host, self.port)
+        self.client.loop_forever()
```
Comment (serroba) [inline app/mqtt_client.py:29]: this answers my previous question.

File: .github/workflows/ci.yaml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+
+    steps:
```
Comment (serroba) [inline .github/workflows/ci.yaml:13]: we will need an step to upload the docker image.
It's cool if you want to do that in a separate PR

**Review by serroba — CHANGES_REQUESTED** [review]
So, the only request I have for you here is to ensure you add the newlines in all the files missing that.

I have a few comments, but I think those can be addressed as separate unit of work to keep the changes not bigger than what they are.

File: main.py (hunk 4 above)
Comment (pieper126) [reply main.py:30]: yes this a blocking operation it will keep looping forever till the container is shutdown.

**Review by serroba — APPROVED** [review]
Let's address the comments in a separate PR
